### PR TITLE
TR with bounds

### PR DESCRIPTION
**Pull request by geoffroyleconte**
Outdates https://github.com/JuliaSmoothOptimizers/RegularizedOptimization.jl/pull/81.

### CODE REVIEW

File: src/TR_alg.jl [hunk 1]
```diff
@@ -191,11 +201,11 @@ function TR(
 
     if η2 ≤ ρk < Inf
       Δk = max(Δk, γ * sNorm)
-      set_radius!(ψ, Δk)
```
Comment (dpo) [inline src/TR_alg.jl:194]: Why did this disappear? The radius should be updated here (not when eta1 ≤ rhok < eta2).

**Review by dpo — CHANGES_REQUESTED** [review]

File: src/TR_alg.jl (hunk 1 above)
Comment (geoffroyleconte) [reply src/TR_alg.jl:194]: $\eta_1 \le \eta_2$, so it will be updated anyways just a few lines below.

Comment (dpo) [reply src/TR_alg.jl:194]: I see. I think it's less clear though. When I read it, I have the impression that there is a bug.

**Review by dpo — APPROVED** [review]
Great, thanks!

**Pull request merged**